src/objects/core/zcl_abapgit_serialize.clas.abap

```diff
@@ -224,7 +226,14 @@ CLASS zcl_abapgit_serialize IMPLEMENTATION.
       iv_language         = io_dot_abapgit->get_main_language( )
       ii_log              = ii_log
       iv_force_sequential = lv_force ).
-    APPEND LINES OF lt_found TO ct_files.
+* transfer one by one and delete from original table as APPEND LINES OF
+* leads to doubling memory usage during execution of statement which
+* could result in SYSTEM_NO_ROLL dump
+    LOOP AT lt_found ASSIGNING <ls_found>.
+      lv_index = sy-tabix.
+      APPEND <ls_found> TO ct_files.
+      DELETE lt_found INDEX lv_index.
+    ENDLOOP.
```

Comment:
Removing row from table doesn't actually help with the memory.
The allocated memory for table lt_found stays the same even after you delete the row from the table.

If you really want to delete the row and save some memory you would have to exit the loop (maybe after you have deleted 1000 rows), copy the table to new variable (which will allocate new memory space, but only for the remaining rows), free the old table and then continue with the loop.

Using delete without this is pointless and provides just an illusion of better memory use.

Reply:
To correct myself:
Deleting rows from the table has effect on memory, only after the table has been fully deleted.
I have run a simple test. Create a stringtab and fill it with numbers from 1 to 100000.
You can check the memory usage using cl_abap_memory_utilities=>get_memory_size_of_object( ).
I delete the rows from the table using index.

Here are the results:
![Full table](https://user-images.githubusercontent.com/17530797/118300827-a47a0200-b4e2-11eb-98d2-20ca047acb9a.png)
![Half deleted table](https://user-images.githubusercontent.com/17530797/118300856-ac39a680-b4e2-11eb-9c83-9dfb7307b97a.png)
![Last record still in table](https://user-images.githubusercontent.com/17530797/118301031-e5721680-b4e2-11eb-9bf0-023cda4dd16d.png)
![Empty table](https://user-images.githubusercontent.com/17530797/118301048-edca5180-b4e2-11eb-9d1d-0afa9b24b080.png)

As you can see the sizeof_used gradually drops as expected.
But sizeof_alloc is near the end greater by more than a half.
It drops to almost nothing after the last row is deleted, but you can do a free after the loop with same effect.

Reply:
in the debugger, try triggering the garbage collector manually?
![image](https://user-images.githubusercontent.com/5888506/118302243-43ebc480-b4e4-11eb-94ec-a351743715b1.png)

Reply:
I tried manually in the debugger and using cl_abap_memory_utilities=>do_garbage_collection( ).
None seems to have an effect, allocated memory stays still same.

Reply:
And just one more correction from my side...
Exiting loop at 50000 rows deleted and creating new variable (using DATA(new_table) = old_table.) allocates the same memory as was for old_table (the bigger number) which is not what we want.
So my workaround is not working.

However what works is exporting the table to shared memory, freeing it and importing it back (done after 50000 rows were deleted).
Before export:
![before_export](https://user-images.githubusercontent.com/17530797/118306129-43a1f800-b4e9-11eb-89a0-1889d4218341.png)
After import:
![after_import](https://user-images.githubusercontent.com/17530797/118306138-44d32500-b4e9-11eb-8b92-107fdfc949d7.png)

Reply:
Well, having no deeper knowledge on memory internals, I'd like to show you what I observed during testing of the change in SAPGUI debugger:

![grafik](https://user-images.githubusercontent.com/47646214/118311933-e0b45f00-b4f0-11eb-9c18-b0e01623ae23.png)
-> "Dyn. Memory Object (Used)" and "Total" got quite lower. Based on that I commited the change.

After your the comments I am not sure what these numbers actually mean.... Shall I change it back to to APPEND LINES OF?

Reply:
@abap-weasel is above only for the APPEND LINES OF change? or the full PR scope?

Reply:
That is only for the APPEND LINES OF change (with some reduced package size). The Total goes a little bit up during append and immediatly reduced by delete again.

Regarding full PR scope, I did not find a good way to validate that exactly. I tried to get the numbers by doing some dumps to SMI transaction: 

![grafik](https://user-images.githubusercontent.com/47646214/118314087-ee1f1880-b4f3-11eb-997e-efbafe563557.png)

Reply:
IMHO I would remove the delete statement and I would just do the append.

Reply:
yea, I agree on that

for the ZIP scenario it should fit into memory anyhow, as the memory requirement is doubled later

Reply:
sure, then let's get back to original APPEND version